python/taichi/_kernels.py

```diff
@@ -179,6 +182,19 @@ def ext_arr_to_matrix(arr: ndarray_type.ndarray(), mat: template(),
                     mat[I][p, q] = arr[I, p, q]
 
 
+@kernel
+def ext_arr_from_vulkan_layout_to_normal_layout(
```

Comment:
Here we should add more comments to explain the purpose of this kernel's existence. And we should indicate that this kernel is only for internal usage.